File: 0658-find-k-closest-elements/0658-find-k-closest-elements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        
        n = len(arr)
        left, right = 0, n - 1
        
        while right - left >= k:
            
            if abs(arr[left] - x) > abs(arr[right] - x):
                left += 1
            else:
                right -= 1
                
        return arr[left:right + 1]
                
                

File: 0658-find-k-closest-elements/0658-find-k-closest-elements.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @param {number} x
 * @return {number[]}
 */
var findClosestElements = function(arr, k, x) {
    let n = arr.length 
    
    let left = 0
    let right = n - 1
    
    while (right - left >= k){
        
        if (Math.abs(arr[left] - x) > Math.abs(arr[right] - x)){
            left += 1
        }
        else{
            right -= 1
        }
    }
    
    return arr.slice(left, right + 1)
};